From e1e60471ae64a049fb4b2793aee7c96a9ee0024a Mon Sep 17 00:00:00 2001
From: Hariharan K <quic_harihk@quicinc.com>
Date: Mon, 5 Feb 2024 16:42:21 +0530
Subject: [PATCH] drivers: firmware: qcom: Updating owner id

While using the __qcom_scm_is_call_available()
service to find the support for
QCOM_SCM_IS_TZ_LOG_ENCRYPTED(0xb), the owner ID
needs to be 0x32.

Change-Id: Ifce4dee1812fc18c33f3f5bbfbcc0a3e60e753e6
Signed-off-by: Hariharan K <quic_harihk@quicinc.com>
---
 drivers/firmware/qcom_scm.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/firmware/qcom_scm.c b/drivers/firmware/qcom_scm.c
index 118e17fa4637..1b0718df9d7a 100644
--- a/drivers/firmware/qcom_scm.c
+++ b/drivers/firmware/qcom_scm.c
@@ -293,7 +293,11 @@ static bool __qcom_scm_is_call_available(struct device *dev, u32 svc_id,
 	switch (__get_convention()) {
 	case SMC_CONVENTION_ARM_32:
 	case SMC_CONVENTION_ARM_64:
-		desc.args[0] = SCM_SMC_FNID(svc_id, cmd_id) |
+		if (cmd_id == QCOM_SCM_IS_TZ_LOG_ENCRYPTED)
+			desc.args[0] = SCM_SMC_FNID(svc_id, cmd_id) |
+				(ARM_SMCCC_OWNER_TRUSTED_OS << ARM_SMCCC_OWNER_SHIFT);
+		else
+			desc.args[0] = SCM_SMC_FNID(svc_id, cmd_id) |
 				(ARM_SMCCC_OWNER_SIP << ARM_SMCCC_OWNER_SHIFT);
 		break;
 	case SMC_CONVENTION_LEGACY:
-- 
2.34.1

